packages/mobile/lib/core/router/onboarding-router.ts

```diff
@@ -0,0 +1,164 @@
+    constructor() {
+        super(getInitialRoute(), onboardingRoute)
+    }
+
+    next(): void {
+        let nextRoute: OnboardingRoute
+
+        const currentRoute = get(this.routeStore)
+        switch (currentRoute) {
+            case OnboardingRoute.AppSetup: {
+                const _onboardingProfile = get(onboardingProfile)
+                if (_onboardingProfile?.isDeveloperProfile) {
+                    nextRoute = OnboardingRoute.NetworkSetup
+                } else {
+                    nextRoute = OnboardingRoute.ProfileSetup
+                }
+                break
+            }
+            case OnboardingRoute.NetworkSetup:
+                nextRoute = OnboardingRoute.ProfileSetup
+                break
+            case OnboardingRoute.ProfileSetup: {
+                const _onboardingProfile = get(onboardingProfile)
+                if (!_onboardingProfile?.mustVisitProfileName) {
+                    const profileType = _onboardingProfile?.type
+                    if (profileType === ProfileType.Software) {
+                        const profileSetupType = _onboardingProfile?.setupType
+                        if (profileSetupType === ProfileSetupType.New) {
+                            nextRoute = OnboardingRoute.StrongholdSetup
+                        } else {
+                            const profileRecoveryType = _onboardingProfile?.recoveryType
+                            if (profileRecoveryType === ProfileRecoveryType.Stronghold) {
+                                nextRoute = OnboardingRoute.StorageProtectionSetup
+                            } else {
+                                nextRoute = OnboardingRoute.StrongholdSetup
+                            }
+                        }
+                    } else {
+                        nextRoute = OnboardingRoute.StorageProtectionSetup
+                    }
+                } else {
+                    nextRoute = OnboardingRoute.ProfileRecovery
+                }
+                break
+            }
+            case OnboardingRoute.StorageProtectionSetup: {
+                const _onboardingProfile = get(onboardingProfile)
+                const profileType = _onboardingProfile?.type
+                if (profileType === ProfileType.Ledger) {
+                    nextRoute = OnboardingRoute.LedgerSetup
+                } else {
+                    const profileSetupType = _onboardingProfile?.setupType
+                    if (profileSetupType === ProfileSetupType.Claimed) {
+                        nextRoute = OnboardingRoute.ShimmerClaiming
+                    } else if (profileSetupType === ProfileSetupType.New) {
+                        nextRoute = OnboardingRoute.ProfileBackup
+                    } else {
+                        const profileRecoveryType = _onboardingProfile?.recoveryType
+                        if (
+                            profileRecoveryType === ProfileRecoveryType.Stronghold ||
```

Comment:
Desktop uses `if (profileRecoveryType === ProfileRecoveryType.Stronghold) {` in this line, do we want to keep it as is now or use the same logic as desktop?

Reply:
I guess we do not need the if-else at this point since `ProfileBackupRoute.BackupStronghold` is not needed, at least not yet. I would just set the Congratulations when the profile type is not "claimed" or "new" at this point.